cmd/main.go

```diff
@@ -2,41 +2,61 @@ package main
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 	"os"
 
 	"github.com/shashkomari/SnakeGame/internal/snakegame"
 )
 
+const kTitle = "config.json"
+
 type Config struct {
 	Board struct {
 		Hight int `json:"hight"`
 		Wight int `json:"width"`
 	} `json:"board"`
 }
 
-func LoadConfiguration(filename string) (Config, error) {
+func LoadConfiguration(filename string) (*Config, error) {
 	var config Config
 
 	configFile, err := os.Open(filename)
 
-	if err != nil {
+	if errors.Is(err, os.ErrNotExist) {
 		configFile, err = os.Create(filename)
+		if err != nil {
+			return nil, err
+		}
+		defer configFile.Close()
+
 		jsonFile := json.NewEncoder(configFile)
 		config.Board.Hight = 10
 		config.Board.Wight = 20
 		err = jsonFile.Encode(&config)
+		if err != nil {
+			return nil, err
+		}
 	} else {
+		if err != nil {
+			return nil, err
+		}
 		defer configFile.Close()
+
 		jsonParser := json.NewDecoder(configFile)
 		err = jsonParser.Decode(&config)
+		if err != nil {
+			return nil, err
+		}
 	}
-	return config, err
+	return &config, nil
 }
 
 func main() {
-	config, _ := LoadConfiguration("config.json")
-
+	config, err := LoadConfiguration(kTitle)
+	if err != nil {
+		panic(err)
```

Comment:
It's not a fault of the program, it's possible behavior based on the system.
Write a log instead of panic.

Reply:
Changed